### ISSUE TITLE
Consider using `nilify_blanks` instead of verbose code

### ISSUE THREAD

**Issue by lentschi** [foodcoops/foodsoft#1107]
Consider using [nilify_blanks](https://github.com/rubiety/nilify_blanks) instead of [current overly verbose code](https://github.com/foodcoopsat/foodsoft_hackathon/blob/bb044e6d2b80c838b1c083114e17729f5537e4a2/app/models/article_version.rb#L75).

**Comment by wvengen** [comment]
Note that more recent versions of Rails have [`normalizes`](https://api.rubyonrails.org/classes/ActiveRecord/Normalization.html#method-i-normalize_attribute), see also [section in the Rails v7.1 release notes](https://guides.rubyonrails.org/v7.2.0/7_1_release_notes.html#add-activerecord-base-normalizes).

**Comment by RayOei** [comment]
Had a quick peek: `nilify_blanks` hasn't been updated for some years, so I presume it is no longer maintained.
The `normalize_attribute` seems a way better candidate and also adds options to clean up like with #1166

**Comment by lentschi** [comment]
> The normalize_attribute seems a way better candidate

Good point. Maybe you could integrate this in [the Rails 7.2 PR](https://github.com/foodcoops/foodsoft/pull/1146), @henning-unicode? The code should be (untested!):

```ruby
# ArticleVersion:
normalizes :unit, :supplier_order_unit, :group_order_unit, :price_unit, :billing_unit, :minimum_order_quantity, with: -> { _1.presence }

# Supplier:
normalizes :supplier_remote_source, :shared_sync_method, with: -> { _1.presence }
```

If this works in your branch, you may just close this issue.
